frontend/src/components/Pages/MissionHistoryPage/MissionHistoryView.tsx

```diff
@@ -23,41 +27,75 @@ const StyledLoading = styled.div`
 export function MissionHistoryView({ refreshInterval }: RefreshProps) {
     const pageSize: number = 10
 
-    const completedStatuses = useMemo(
-        () => [
-            MissionStatus.Aborted,
-            MissionStatus.Cancelled,
-            MissionStatus.Successful,
-            MissionStatus.PartiallySuccessful,
-            MissionStatus.Failed,
-        ],
-        []
-    )
-    const [completedMissions, setCompletedMissions] = useState<Mission[]>([])
+    const [filteredMissions, setFilteredMissions] = useState<Mission[]>([])
     const [paginationDetails, setPaginationDetails] = useState<PaginationHeader>()
-    const [currentPage, setCurrentPage] = useState<number>()
     const [isLoading, setIsLoading] = useState<boolean>(true)
+    const [isResettingPage, setIsResettingPage] = useState<boolean>(false)
 
-    const updateCompletedMissions = useCallback(() => {
-        const page = currentPage ?? 1
-        BackendAPICaller.getMissions({ pageSize: pageSize, pageNumber: page, orderBy: 'EndTime desc, Name' }).then(
-            (paginatedMissions) => {
-                setPaginationDetails(paginatedMissions.pagination)
-                setCompletedMissions(paginatedMissions.content.filter((m) => completedStatuses.includes(m.status)))
-                setIsLoading(false)
+    const {
+        page,
+        switchPage,
+        missionName,
+        statuses,
+        robotName,
+        tagId,
+        inspectionTypes,
+        minStartTime,
+        maxStartTime,
+        minEndTime,
+        maxEndTime,
+    } = useMissionFilterContext()
+
+    const updateFilteredMissions = useCallback(() => {
+        BackendAPICaller.getMissions({
+            statuses: statuses,
+            nameSearch: missionName,
+            robotNameSearch: robotName,
+            tagSearch: tagId,
+            inspectionTypes: inspectionTypes,
+            minStartTime: minStartTime,
+            maxStartTime: maxStartTime,
+            minEndTime: minEndTime,
+            maxEndTime: maxEndTime,
+            pageSize: pageSize,
+            pageNumber: page ?? 1,
+            orderBy: 'EndTime desc, Name',
+        }).then((paginatedMissions) => {
+            setFilteredMissions(paginatedMissions.content)
+            setPaginationDetails(paginatedMissions.pagination)
+            if (page > paginatedMissions.pagination.TotalPages && paginatedMissions.pagination.TotalPages > 0) {
+                switchPage(paginatedMissions.pagination.TotalPages)
+                setIsResettingPage(true)
             }
-        )
-    }, [completedStatuses, currentPage, pageSize])
+            setIsLoading(false)
+        })
+    }, [
+        page,
```

Comment:
If you do end up switching to having a single state object in the context then you would only need to check for an update to that object, since here we don't care which specific value is updated